src/openrct2/world/MapGen.cpp

```diff
@@ -349,7 +349,8 @@ static void mapgen_place_trees()
         pos = availablePositions[i];
 
         int32_t type = -1;
-        auto* surfaceElement = map_get_surface_element_at(pos.x, pos.y);
+        // TODO: This is Converting CoordsXY to TileCoordsXY directly without /32
+        auto* surfaceElement = map_get_surface_element_at(TileCoordsXY{ pos.x, pos.y });
```

Comment:
Please comment on this. `pos` is a `CoordsXY` object, so I'm thinking this is a bug.

Reply:
L307 and L309 are incorrect this is a `TileCoordsXY` object but really we should be modifying this function to make this `CoordsXY`